### PR TITLE
Ex04

### PR DESCRIPTION
**Pull request by gazelleazadi**
Please review my assignment.

### CODE REVIEW

File: src/ghazal_azadi_ex/ex04/walker.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        self.steps = 0
+
+    def is_at_home(self):
+
+        if self.position == self.home_position:
+            self.state = True
+        else:
+            self.state = False
+
+        return self.state
```
Comment (huynhngoc) [inline src/ghazal_azadi_ex/ex04/walker.py]: `return self.position == self.home_position`

File: src/ghazal_azadi_ex/ex04/walker.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import random
+
+
+class Walker:
+
+    def __init__(self, position, home_position):
+        self.position = position
+        self.home_position = home_position
+        self.state = True
```
Comment (huynhngoc) [inline src/ghazal_azadi_ex/ex04/walker.py]: remove this redundant attribute

File: src/ghazal_azadi_ex/ex04/walker.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        return self.position
+
+    def get_steps(self):
+        return self.steps
+
+    def walking_process(self):
+        while not self.is_at_home():
+            Walker.move(self)
+        return Walker.get_steps(self)
```
Comment (huynhngoc) [inline src/ghazal_azadi_ex/ex04/walker.py]: This is wrong.
Instead, create the walking_process function outside this class (as an independent function)

Comment (gazelleazadi) [reply src/ghazal_azadi_ex/ex04/walker.py]: why should not this function be part of the class?

**Pull request merged**